### PR TITLE
feat(cli): Add --no-git-restore flag to resume sessions without applying git changes

### PR DESCRIPTION
**Pull request by marius-kilocode**
## Summary

Adds a new `--no-git-restore` CLI option that allows users to resume sessions without applying git state changes (checkout, stash, patch application).

## Motivation

When resuming a session, the CLI normally restores the git state from when the session was saved. However, there are cases where users want to continue a conversation without modifying their working directory:
- The user has made local changes they don't want to lose
- The user wants to review the session on a different branch
- The user wants to continue the conversation without the original git context

## Usage

```bash
# Resume a session without applying git changes
kilocode --session abc123 --no-git-restore
```

## Changes

- **CLI**: Added `--no-git-restore` option with validation (requires `--session`)
- **SessionManager**: Added `RestoreSessionOptions` interface with `skipGitRestore` and `throwOnGitError` properties
- **SessionLifecycleService**: Skip git state restoration when flag is set, with logging
- **GitStateService**: Added `throwOnError` option to fail fast on git restore errors (enabled by default)
- **forkSession**: Always restores git state (required to continue the forked session properly)
- **Tests**: Added comprehensive tests covering all edge cases

## Behavior Summary

| Command | Git Restore | Error Handling |
|---------|-------------|----------------|
| `--session` | Enabled | Fail fast (throws on checkout/patch errors) |
| `--session --no-git-restore` | Skipped | N/A |
| `--fork` | Always enabled | Fail fast (throws on checkout/patch errors) |

## Test Plan

- [x] All existing tests pass
- [x] New tests cover `skipGitRestore` option behavior
- [x] New tests cover `throwOnGitError` option behavior
- [x] Type checking passes

### CODE REVIEW

**Comment by marius-kilocode** [comment]
@iscekic I think i need to change this. It doesn't make sense to use fork with `--no-git-restore`

File: src/shared/kilocode/cli-sessions/core/SessionManager.ts [hunk 1]
```diff
@@ -206,9 +217,13 @@ export class SessionManager {
...
 	 * Restores a session by ID from the cloud.
 	 * Delegates to SessionLifecycleService.
+	 *
+	 * @param sessionId - The session ID to restore
+	 * @param rethrowError - Whether to rethrow errors (default: false)
+	 * @param options - Optional restore options (e.g., skipGitRestore)
 	 */
-	async restoreSession(sessionId: string, rethrowError = false): Promise<void> {
-		return this.lifecycleService.restoreSession(sessionId, rethrowError)
+	async restoreSession(sessionId: string, rethrowError = false, options?: RestoreSessionOptions): Promise<void> {
```
Comment (iscekic) [inline src/shared/kilocode/cli-sessions/core/SessionManager.ts]: Lets put `rethrowError` in the options

File: src/shared/kilocode/cli-sessions/core/SessionManager.ts [hunk 2]
```diff
@@ -230,9 +245,13 @@ export class SessionManager {
...
 	 * Forks a session by share ID or session ID.
 	 * Delegates to SessionLifecycleService.
+	 *
+	 * @param shareOrSessionId - The share ID or session ID to fork
+	 * @param rethrowError - Whether to rethrow errors (default: false)
+	 * @param options - Optional restore options (e.g., skipGitRestore)
 	 */
-	async forkSession(shareOrSessionId: string, rethrowError = false): Promise<void> {
-		return this.lifecycleService.forkSession(shareOrSessionId, rethrowError)
+	async forkSession(shareOrSessionId: string, rethrowError = false, options?: RestoreSessionOptions): Promise<void> {
```
Comment (iscekic) [inline src/shared/kilocode/cli-sessions/core/SessionManager.ts]: Same here

**Review by iscekic — CHANGES_REQUESTED** [review]
LGTM, minor changes needed
